### ISSUE TITLE
Documentation of common region vs. model-native region

### ISSUE THREAD

**Issue by danielhuppmann** [IAMconsortium/nomenclature#145]
Per this comment by @luciecastella in https://github.com/IAMconsortium/nomenclature/issues/144#issuecomment-1210296974_, it would be useful to have an own section explaining the difference between a "common" region vs. a "model_native" region. Could be an own new sub-page of the user guide or in the Region section of the "codelist" page.

**Comment by danielhuppmann** [comment]
Looking at the docs again, I think pulling our the "variable" and "region" sections from `codelist.rst` to own pages (and only keep the generic info on that page, plus links to the detailed pages for the more complex dimensions) is the best approach...

**Comment by phackstock** [comment]
That is what you have implemented in #140 right (minus the links on the codelist.rst to "variable" and "region")? It's just not in the "stable" part of the docs yet since it's not part of the latest release. In the "latest" version (https://nomenclature-iamc.readthedocs.io/en/latest/user_guide.html) we have that split. Maybe we should think about another release once #144 is merged.

**Comment by danielhuppmann** [comment]
Right, I just looked at the default (stable) docs page on RTD. Good to know that at least my thinking is consistent over time... 😜

And yes, @luciecastella, up to you to do a release after tackling #144!
